tools-local/Microsoft.ML.InternalCodeAnalyzer/BestFriendAnalyzer.cs

```diff
@@ -17,21 +17,21 @@ public sealed class BestFriendAnalyzer : DiagnosticAnalyzer
         private const string Category = "Access";
         internal const string DiagnosticId = "MSML_NoBestFriendInternal";
 
-        private string Title => "Cross-assembly internal access requires referenced item to have " + _attributeName + " attribute.";
-        private string Format => "Access of '{0}' is a cross assembly internal " +
+        private const string Title = "Cross-assembly internal access requires referenced item to have " + AttributeName + " attribute.";
+        private const string Format = "Access of '{0}' is a cross assembly internal " +
             "reference, and the declaring assembly wants these accesses to be on something " +
-            "with the attribute " + _attributeName + ".";
-        private string Description =>
+            "with the attribute " + AttributeName + ".";
+        private const string Description =
             "The identifier indicated is defined as an internal member of an assembly that has the " +
-            _assemblyAttributeName + " assembly-level attribute set. Even with friend access to that " +
-            "assembly, such a usage requires that the item have the " + _attributeName + " on it.";
+            AssemblyAttributeName + " assembly-level attribute set. Even with friend access to that " +
+            "assembly, such a usage requires that the item have the " + AttributeName + " on it.";
 
-        private DiagnosticDescriptor Rule =>
+        private static DiagnosticDescriptor Rule =
             new DiagnosticDescriptor(DiagnosticId, Title, Format, Category,
                 DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
 
-        private string _attributeName;
-        private string _assemblyAttributeName;
+        private const string AttributeName = "Microsoft.ML.BestFriendAttribute (or Microsoft.ML.Runtime.Internal.CpuMath.BestFriendAttribute)";
```

Comment:
You could just say `BestFriendAttribute` and `WantsToBeBestFriendsAttribute` in the message. I don't think it needs to be fully qualified, especially since it makes it more confusing to the user IMO.